bitnami/thanos/templates/query/ingress.yaml

```
@@ -9,8 +9,10 @@ kind: Ingress
 metadata:
   name: {{ include "thanos.query.fullname" . }}
   namespace: {{ include "common.names.namespace" . }}
-  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
-    app.kubernetes.io/component: query
+  {{- if or .Values.query.ingress.labels .Values.commonLabels }}
+  {{- $labels := include "common.tplvalues.merge" ( dict "values" ( list .Values.query.ingress.labels .Values.commonLabels ) "context" . ) }}
+  labels: {{- include "common.tplvalues.render" ( dict "value" $labels "context" $) | nindent 4 }}
+  {{- end }}  
```

Labels should be set regardless they're `.Values.query.ingress.labels` OR `.Values.commonLabels` are set:
```suggestion
  {{- $labels := include "common.tplvalues.merge" ( dict "values" ( list .Values.query.ingress.labels .Values.commonLabels ) "context" . ) }}
  labels: {{- include "common.labels.standard" ( dict "customLabels" $labels "context" $ ) | nindent 4 }}
    app.kubernetes.io/component: query
```

---

Thanks, you are right.
I've applied the proposed changes